packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -286,22 +286,23 @@ class Dart2WasmTarget extends Dart2WebTarget {
     if (compileResult.exitCode != 0) {
       throw Exception(_collectOutput(compileResult));
     }
-    if (compilerConfig.runWasmOpt) {
+    if (compilerConfig.wasmOpt.runIt) {
       final String wasmOptBinary = artifacts.getArtifactPath(
         Artifact.wasmOptBinary,
         platform: TargetPlatform.web_javascript
       );
       final File optimizedOutput = environment.buildDir.childFile('main.dart.wasm');
       final List<String> optimizeArgs = <String>[
         wasmOptBinary,
-        '-all',
+        '--all-features',
```

Comment:
using un-abbreviated names. Easier to search, etc.

Also sorting these

Reply:
The order of the flags are important, and we need `-O3` twice in those particular positions because there are some passes we need to run before and after other passes.